frontends/p4/fromv1.0/programStructure.cpp

```diff
@@ -1090,9 +1104,17 @@ CONVERT_PRIMITIVE(bit_xnor) {
 CONVERT_PRIMITIVE(add) {
     OPS_CK(primitive, 3);
     auto args = convertArgs(structure, primitive);
+    bool isSaturated = false;
+    for (auto a : args)
+        isSaturated = isSaturated || isSaturatedField(a);  // any of the fields
```

Comment:
would

    bool isSaturated = std::any_of(args.begin(), args.end(), isSaturatedField);

be simpler/clearer?